.github/workflows/build.yml

```diff
@@ -0,0 +1,34 @@
+name: Build
+
+on:
+  push:
+    branches: [master]
+
+jobs:
+  generate:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x]
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: npm install and build
+      run: |
+        npm ci
+        npm start
+      env:
+        CI: true
+    - name: Deploy
+      uses: peaceiris/actions-gh-pages@v2
```

Comment:
It all looks great, just want to confirm that we're using `actions-gh-pages` just as a tool for publishing to branch, and it has nothing to do with [GitHub Pages](https://pages.github.com/) in our specific case, right?

Reply:
Yeah, the action is conventionally used to publish static sites to a `gh-pages` branch, but it does not actually do any GitHub Pages-specific stuff despite its name and works for our use case. Here's the [source code](https://github.com/peaceiris/actions-gh-pages/blob/master/entrypoint.sh) if you are interested.